### PR TITLE
*: add builtin function FROM_DAYS

### PR DESCRIPTION
**Pull request by Tratar**
Implemented FROM_DAYS builtin function.

### CODE REVIEW

**Comment by shenli** [comment]
@Tratar Thanks for your PR! Please solve the conflict.
@XuHuaiyu PTAL

**Comment by shenli** [comment]
@tiancaiamao PTAL

File: expression/builtin_time_test.go [hunk 1]
```diff
@@ -471,6 +471,36 @@ func (s *testEvaluatorSuite) TestStrToDate(c *C) {
 	}
 }
 
+func (s *testEvaluatorSuite) TestFromDays(c *C) {
```
Comment (XuHuaiyu) [inline expression/builtin_time_test.go]: Hi, please add some test cases for string and negative numbers.
like "149a" "366a" "abc"

**Comment by tiancaiamao** [comment]
Hi, @Tratar 
We have a `getDateFromDaynr ` https://github.com/pingcap/tidb/blob/master/util/types/mytime.go#L313, you can use it instead of `calcTimefromDays`.

File: expression/builtin_time_test.go (hunk 1 above)
Comment (Tratar) [inline expression/builtin_time_test.go]: Done

**Comment by Tratar** [comment]
Fixed, please review.

**Comment by shenli** [comment]
@Tratar CI failed:
util/types/time.go:1134:1: exported function TimeFromDays should have comment or be unexported
make: *** [check] Error 1

**Comment by shenli** [comment]
@XuHuaiyu @tiancaiamao PTAL

File: util/types/mytime.go [hunk 2]
```diff
@@ -149,6 +149,10 @@ func calcTimeDiff(t1, t2 TimeInternal, sign int) (seconds, microseconds int, neg
 	return
 }
 
+func isYearLeap(year int) bool {
```
Comment (tiancaiamao) [inline util/types/mytime.go]: You could remove this function if it's not referenced anywhere.

File: util/types/time.go [hunk 3]
```diff
@@ -1131,6 +1131,23 @@ func ParseDateFromNum(num int64) (Time, error) {
 	return ParseTimeFromNum(num, mysql.TypeDate, MinFsp)
 }
 
+func TimeFromDays(num int64) Time {
```
Comment (tiancaiamao) [inline util/types/time.go]: Add comment for this function to make the lint check happy.

**Comment by Tratar** [comment]
Thanks much. Fixed, please review.

**Comment by tiancaiamao** [comment]
CI failed. 
You cant run `make check` twice, until golint don't report any error. @Tratar

**Comment by tiancaiamao** [comment]
LGTM
PTAL @XuHuaiyu

**Comment by XuHuaiyu** [comment]
LGTM
Thank you for your job.

**Pull request merged**